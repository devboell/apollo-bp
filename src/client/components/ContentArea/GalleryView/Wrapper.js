import styled from 'styled-components'

const Wrapper = styled.div.attrs({
  'data-testid': 'galleryView',
})`
`

export default Wrapper
