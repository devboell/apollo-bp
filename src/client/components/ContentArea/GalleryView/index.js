import React from 'react'
import PropTypes from 'prop-types'
import { compose, uniq, pluck } from 'ramda'

import GenreChooser from '../GenreChooser'
import GalleryThumbs from '../GalleryThumbs'
import Wrapper from './Wrapper'

const artistGenres =
  compose(
    uniq,
    pluck('genre'),
  )

export class GalleryView extends React.Component {
  componentDidMount() {
    const {
      artist,
      selectedGenres,
      onSetSelectedGenres,
    } = this.props
    if (selectedGenres.length === 0) {
      onSetSelectedGenres(artist.path, artistGenres(artist.paintings))
    }
  }

  render() {
    const {
      artist: {
        path,
        paintings,
      },
      selectedGenres,
      onSelectGenre,
      onDeselectGenre,
    } = this.props

    const genreFilter = painting => selectedGenres.includes(painting.genre)

    return (
      <Wrapper>
        <GenreChooser
          genres={artistGenres(paintings)}
          selectedGenres={selectedGenres}
          onSelectGenre={onSelectGenre(path)}
          onDeselectGenre={onDeselectGenre(path)}
        />
        <GalleryThumbs
          artistPath={path}
          paintings={paintings.filter(genreFilter)}
        />
      </Wrapper>
    )
  }
}

GalleryView.propTypes = {
  artist: PropTypes.shape({
    path: PropTypes.string,
    paintings: PropTypes.array,
  }),
  selectedGenres: PropTypes.arrayOf(PropTypes.string),
  onSetSelectedGenres: PropTypes.func.isRequired,
  onSelectGenre: PropTypes.func.isRequired,
  onDeselectGenre: PropTypes.func.isRequired,
}

GalleryView.defaultProps = {
  artist: {},
  selectedGenres: [],
}

export default GalleryView

