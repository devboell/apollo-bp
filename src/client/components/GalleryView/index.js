import React from 'react'
import { PropTypes } from 'prop-types'

import GalleryItem from './GalleryItem'
import Wrapper from './Wrapper'


export const GalleryView = ({ paintings, artistPath }) =>
  <Wrapper>
    {paintings.map(painting =>
      <GalleryItem
        key={`painting-${painting.name}`}
        artistPath={artistPath}
        {...{ painting }}
      />,
    )}
  </Wrapper>

GalleryView.propTypes = {
  paintings: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string })).isRequired,
  artistPath: PropTypes.string.isRequired,
}

export default GalleryView
