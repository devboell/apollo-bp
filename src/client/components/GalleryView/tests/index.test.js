import React from 'react'
import GalleryView from '../index'


const props = {
  artistPath: 'testPath',
  paintings: [
    {
      title: 'painting1',
      name: 'painting1.jpg',
    },
    {
      title: 'painting2',
      name: 'painting2.jpg',
    },
  ],
}

it('GalleryView component snapshot', () => {
  const wrapper = shallow(<GalleryView {...props} />)
  expect(wrapper).toMatchSnapshot()
})
