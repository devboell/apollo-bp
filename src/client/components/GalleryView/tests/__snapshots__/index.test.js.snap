// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryView component snapshot 1`] = `
<styled.div>
  <GalleryItem
    artistPath="testPath"
    painting={
      Object {
        "name": "painting1.jpg",
        "title": "painting1",
      }
    }
  />
  <GalleryItem
    artistPath="testPath"
    painting={
      Object {
        "name": "painting2.jpg",
        "title": "painting2",
      }
    }
  />
</styled.div>
`;
